salt/roster/flat.py

```diff
@@ -44,19 +44,20 @@ def targets(tgt, tgt_type='glob', **kwargs):
     conditioned_raw = {}
     for minion in raw:
         conditioned_raw[str(minion)] = raw[minion]
-    rmatcher = RosterMatcher(conditioned_raw, tgt, tgt_type, 'ipv4')
+    rmatcher = RosterMatcher(conditioned_raw, tgt, tgt_type, 'ipv4', __opts__)
     return rmatcher.targets()
 
 
 class RosterMatcher(object):
     '''
     Matcher for the roster data structure
     '''
-    def __init__(self, raw, tgt, tgt_type, ipv='ipv4'):
+    def __init__(self, raw, tgt, tgt_type, ipv='ipv4', opts={}):
```

Comment:
Ah, right, I noticed now what you want to do. I think it depends where you reuse the class or not in the future. If you plan to call it:
```python
rmatcher = RosterMatcher(raw, tgt, tgt_type, 'ipv6', __opts__)
```
Then your way is would work. However personally I would go the kwargs way as it accepts _only_ named arguments and thus do this:
```python

def __init__(self, raw, tgt, tgt_type, ipv='ipv4', **kwargs):
    self.opts = kwargs.get('opts', {}).copy()
```
In either way you still need to fix a bug, as in your case you create only a reference to an `__opts__`, and so if you change anything in `self.opts`, you also change inside `__opts__`. IIRC, this is not what you want (otherwise you'd better_just_ access `__opts__` directly to avoid confusion).